cmd/proxy/wsl_integration_linux.go

```diff
@@ -28,20 +31,35 @@ import (
 )
 
 var (
-	debug        bool
-	upstreamAddr string
-	listenAddr   string
+	debug            bool
+	upstreamAddr     string
+	listenAddr       string
+	keyFile          string
+	certFile         string
+	rootCA           string
+	upstreamKeyFile  string
+	upstreamCertFile string
 )
 
 const (
-	k8sAPI            = "https://192.168.1.2:6443"
-	defaultListenAddr = "127.0.0.1:6443"
+	k8sAPI               = "https://192.168.1.2:6443"
+	defaultListenAddr    = "127.0.0.1:6443"
+	kubeAPIServerKeyFile = "/var/lib/rancher/k3s/server/tls/serving-kube-apiserver.key"
+	kubeAPIServerCrtFile = "/var/lib/rancher/k3s/server/tls/serving-kube-apiserver.crt"
+	rootCACrtFile        = "/var/lib/rancher/k3s/server/tls/server-ca.crt"
+	clientAdminKeyFile   = "/var/lib/rancher/k3s/server/tls/client-admin.key"
+	clientAdminCrtFile   = "/var/lib/rancher/k3s/server/tls/client-admin.crt"
 )
 
 func main() {
 	flag.BoolVar(&debug, "debug", false, "enable additional debugging")
 	flag.StringVar(&upstreamAddr, "upstream-addr", k8sAPI, "The upstream server's address.")
 	flag.StringVar(&listenAddr, "listen-addr", defaultListenAddr, "The server's address in an IP:PORT format.")
+	flag.StringVar(&keyFile, "key-file", kubeAPIServerKeyFile, "TLS private key file for downstream connection")
+	flag.StringVar(&certFile, "cert-file", kubeAPIServerCrtFile, "TLS certificate file for downstream connection")
+	flag.StringVar(&rootCA, "root-ca-file", rootCACrtFile, "root ca file for upstream connection")
+	flag.StringVar(&upstreamKeyFile, "upstream-key-file", clientAdminKeyFile, "TLS private key file for upstream connection")
+	flag.StringVar(&upstreamCertFile, "upstream-cert-file", clientAdminCrtFile, "TLS certificate file for upstream connection")
```

Comment:
`upstreamKeyFile` and `upstreamCertFile` are client certificates (to be used when talking to upstream); it might be good to mention `client` somewhere in the description (and probably the variable names).

Question: how would I make a connection as a service account, since we're always overriding the client here?

Reply:
> Question: how would I make a connection as a service account, since we're always overriding the client here?

I have created and issue here to discuss this further: https://github.com/rancher-sandbox/rancher-desktop/issues/6553